### PR TITLE
feat(apigw + waf): separate logging enabled from logging profiles

### PR DESCRIPTION
**Pull request by RossMurr4y**
## Description

* Separated out the enabling of logging from the logging profile for the `apigateway` component and `WAF` configurations.

Note: I've used objects for the `apigateway` attributes because the intent is for providers to provide any further control they require at the provider level. A bool would not have been sufficient control here but any further configuration here will be provider specific.

## Motivation and Context

The APIGateway provides several kinds of logs - Access Logs, Execution Logs and also WAF Logs. However they are not always able to be configured the same way. Separating out the concerns of "do you want logs?" from "where do you want them?" (`LoggingProfiles`) and creating an attribute structure likes this allows providers to add further control should they wish through namespaced attributes, whilst remaining clear as to the intent of each attribute.

## How Has This Been Tested?

Local template testing.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions

- [ ] Schema regeneration will be required.

## Checklist:

- [x] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] None of the above.

### CODE REVIEW

File: providers/shared/components/apigateway/id.ftl [hunk 1]
```diff
@@ -75,6 +75,33 @@ object.
...
+                        "Names" : "Execution",
+                        "Children" : [
+                            {
+                                "Names" : "Enabled",
+                                "Description" : "Logging of the API Gateway Execution Logs.",
+                                "Type" : BOOLEAN_TYPE,
+                                "Default" : false
+                            }
+                        ]
+                    },
```
Comment (roleyfoley) [inline providers/shared/components/apigateway/id.ftl:91]: I don't think we should include this one. The execution log is a troubleshooting log and should be mandatory for all deployments. The access logging is optional as that can have significant cost implications

File: providers/shared/components/component.ftl [hunk 2]
```diff
@@ -592,11 +592,18 @@
             "Type" : BOOLEAN_TYPE,
             "Default" : false
         },
+        {
+            "Names" : "EnableLogging",
+            "Description" : "WAF Logs in the OpsData DataBucket.",
+            "Type" : BOOLEAN_TYPE,
+            "Default" : true
+        },
```
Comment (roleyfoley) [inline providers/shared/components/component.ftl:600]: To keep it constistent with the changes above should this be 

```suggestion
        {
            "Names" : "Logging",
            "Children" : [ 
                {
                    "Names" : "Enabled",
                    "Type" : BOOLEAN_TYPE,
                    "Default" : true
                }
            ]
        },
```

File: providers/shared/components/apigateway/id.ftl (hunk 1 above)
Comment (roleyfoley) [reply providers/shared/components/apigateway/id.ftl:91]: If this is removed you could also make it an Object shorter and make it 

AccessLogging with children of Enabled

**Review by roleyfoley — CHANGES_REQUESTED** [review]

File: providers/shared/components/apigateway/id.ftl (hunk 1 above)
Comment (RossMurr4y) [reply providers/shared/components/apigateway/id.ftl]: Thanks, updated in b973911 and 8bff92b

File: providers/shared/components/component.ftl (hunk 2 above)
Comment (RossMurr4y) [reply providers/shared/components/component.ftl]: Updated in bc14a90

Comment (RossMurr4y) [reply providers/shared/components/component.ftl]: @roleyfoley should we also update the `EnableLogging` attr on `apigateway` within the `cloudfront` attribute (i.e `solution.CloudFront.EnableLogging` to `solution.CloudFront.Logging.Enabled` and similarly the one on the CDN component?

**Review by roleyfoley — APPROVED** [review]

**Pull request merged**